### ISSUE TITLE
Segfault in reflect.valueInterfaceUnsafe

### ISSUE THREAD

**Issue by dgryski** [tinygo-org/tinygo#2323]
This code is extracted and minimized from the easyjson test suite:

```
package main

import (
	"fmt"
	"reflect"
)

var arrayValue [3]byte

func main() {
	var test = struct {
		Decoded *[3]byte
	}{
		&arrayValue,
	}

	v1 := reflect.New(reflect.TypeOf(test.Decoded).Elem())
	fmt.Println(v1)
}
```

Both calling `fmt.Println` or `reflect.DeepEqual` as the test was doing causes a segfault in `reflect.valueInterfaceUnsafe`.  Printing out the typecode, value, and flags on entrance to this `valueInterfaceUnsafe` prints the following

```
0x0000000000000925 0x000000011394e060 2
0x0000000000000049 0x000000011394e060 3
0x0000000000000010 0x0000000000000060 3
```

This does not fail on `target=wasi`.

```
0x00000925 0x00016cc0 2
0x00000049 0x00016cc0 3
0x00000010 0x000000c0 3
0x00000010 0x0000006c 3
0x00000010 0x00000001 3
&[0 0 0]
```

**Comment by dgryski** [comment]
`encoding/binary` also fails with trying to set an array element.  I wonder if it's related also?

**Comment by dgryski** [comment]
The problem here seems to be that `reflect.New` is creating an indirect value even though the result type (a `[3]byte`) fits into a  `uintptr`, and then things Get Confused during `Index` when the assumptions don't match up.

**Comment by dgryski** [comment]
(This was the same encoding/binary issue.)

**Comment by aykevl** [comment]
> The problem here seems to be that `reflect.New` is creating an indirect value even though the result type (a `[3]byte`) fits into a `uintptr`, [...]

`reflect.New` always returns a pointer, so this behavior seems correct to me.

**Comment by dgryski** [comment]
Yes.  The issue was more `isIndirect` issues, also fixed by the last commit in #2395 .
